lib/ash/type/type.ex

```diff
@@ -1203,6 +1203,8 @@ defmodule Ash.Type do
   def apply_constraints({:array, type}, term, constraints) when is_list(term) do
     type = get_type(type)
 
+    term = remove_empty_items(term, constraints)
```

Comment:
I don't think we want to remove empty items *before* applying item constraints, because we don't know what will and won't be empty until we do that.